### ISSUE TITLE
Aborting input is not possible, implement SIGINT trap?

### ISSUE THREAD

**Issue by Radiergummi** [fisherman/get#3]
I noticed you cannot abort input, which is inconvenient if you tend to derp around like me. Would it be possible to implement a SIGINT trap to watch for ^C?

**Comment by brj** [comment]
I would really like to have this, but last time I tried:

``` 
trap get_clean INT
```
would still _not_ return the cursor in some situations for no apparent reason, so I gave up. 

Do you know how these trap things work?

**Comment by Radiergummi** [comment]
Nope, that's just what I've tried, but I can't seem to work out the finer details of fish traps.. Did this a few times with bash though, no problems there...

Maybe @ridiculousfish can help here?
